spec/migrations/20160802145938_set_generic_subtype_on_service_template_spec.rb

```diff
@@ -0,0 +1,41 @@
+require_migration
+
+describe SetGenericSubtypeOnServiceTemplate do
+  let(:service_template_stub) { migration_stub(:ServiceTemplate) }
+
+  migration_context :up do
+    it 'sets generic_subtype to custom on generic Service Templates' do
+      st = service_template_stub.create!(:prov_type => 'generic')
+
+      migrate
+
+      expect(st.reload.generic_subtype).to eq('custom')
+    end
+
+    it 'skips non-generic Service Templates' do
+      st = service_template_stub.create!(:prov_type => 'vmware')
+
+      migrate
+
+      expect(st.reload.generic_subtype).to be_nil
+    end
+  end
+
+  migration_context :down do
+    it 'sets generic_subtype to nil on generic Service Templates' do
+      st = service_template_stub.create!(:prov_type => 'generic', :generic_subtype => 'custom')
+
+      migrate
+
+      expect(st.reload.generic_subtype).to be_nil
+    end
+
+    it 'skips non-generic Service Templates' do
+      st = service_template_stub.create!(:prov_type => 'vmware')
+
+      migrate
+
+      expect(st.reload.generic_subtype).to be_nil
```

Comment:
The down migration is setting the `generic_subtype` value to `nil` so this test is not going to catch any issues because the `service_template` created here will already have a nil value.

Instead the test should add a value in `generic_subtype` and verify it does not get cleared.